File: Implementation/Mathematics/BasicMath/python/armstrongNumbers.py
def isArNum(n, s):
    c = 0
    t = n
    while t > 0:
        r = t % 10
        c += r ** s
        t //= 10

    return (c == n)

n = int(input())
s = len(str(n))
print(isArNum(n, s))

File: Implementation/Mathematics/BasicMath/cpp/armstrongNumbers.cpp
#include <bits/stdc++.h>

using namespace std;

int size(int n)
{
    int c = 0;
    while (n)
    {
        ++c;
        n /= 10;
    }

    return c;
}

bool isArNum(int n)
{
    int s = size(n);
    int t = n, c = 0, r;
    while (t)
    {
        r = t % 10;
        c += pow(r, s);
        t /= 10;
    } 

    return (n == c);
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    (isArNum(n)) ? cout << "Yes\n" : cout << "No\n";

    return 0;
}
